infer/resource.go

```diff
@@ -292,8 +292,8 @@ type Annotator interface {
 	// `mypkg:mymodule:MyResource`, in the same way `SetToken` does.
 	AddAlias(module tokens.ModuleName, name tokens.TypeName)
 
-	// Set a deprecation message for the resource, which officially marks it as deprecated.
-	SetResourceDeprecationMessage(message string)
+	// Set a deprecation message for a struct field, which officially marks it as deprecated.
+	Deprecate(i any, message string)
```

Comment:
I'd put an example here.
```suggestion
	// Set a deprecation message for a struct field, which officially marks it as deprecated.
	//	func (*s Struct) Annotated(a Annotator) {
	//		a.Deprecate(&s.Field, "field is deprecated")
	//	}
	//
	// To deprecate a resource, object or function, call Deprecate on the struct itself:
	//
	//	func (*s Struct) Annotated(a Annotator) {
	//		a.Deprecate(&s, "Struct is deprecated")
	//	}
	Deprecate(i any, message string)
```